web/src/components/Header/HeaderRightPanel.tsx

```diff
@@ -171,19 +172,19 @@ function HeaderRightPanel({
               {(!noFilters || isTextFilter) && (goalList.filter((goal) => (
                   goal.content.includes(filterText)
                 )).map((goal) => (
-                  <SearchResultItem text={goal.content} name="comment.svg"/>
+                  <SearchResultItem text={goal.content} name="acorn-logo.svg" onExpandClick={openExpandedView} goalAddress={goal.headerHash}/>
                 )))
               }
               {(!noFilters || isDescriptionFilter) && (goalList.filter((goal) => (
                   goal.description.includes(filterText)
                 )).map((goal) => (
-                  <SearchResultItem text={goal.description} name="comment.svg"/>
+                  <SearchResultItem text={goal.description} name="equalizer.svg" onExpandClick={openExpandedView} goalAddress={goal.headerHash}/>
                 )))
               }
               {(!noFilters || isCommentFilter) && (commentList.filter((comment) => (
                   comment.content.includes(filterText)
                 )).map((comment) => (
-                  <SearchResultItem text={comment.content} name="comment.svg"/>
+                  <SearchResultItem text={comment.content} name="comment.svg" onExpandClick={openExpandedView} goalAddress={comment.headerHash}/>
```

Comment:
for a goalComment, you'll need to refer to a different property, I forget which property, but it should be obvious by looking at the GoalComment data structure (in Rust)